### PR TITLE
Sdl winport merge to develop with tests (part4) 

### PR DESCRIPTION
**Pull request by OHerasym**
Fix protocol_handler tests
Fix transport manager tests
Fix security manager unit tests

### CODE REVIEW

**Comment by OHerasym** [comment]
@anosach-luxoft , @dev-gh, @VVeremjova , @AByzhynar, @Kozoriz, @LuxoftAKutsan, @dtrunov, Please review

File: src/components/security_manager/test/security_query_test.cc [hunk 1]
```diff
@@ -134,7 +134,8 @@ TEST_F(SecurityQueryTest, QueryHeaderCopyConstructor) {
 /*
  * Security QueryHeader shall construct with NULL fields
  */
-TEST_F(SecurityQueryTest, QueryConstructor) {
+// TODO(OHerasym) : out of range exception on Windows platform
```
Comment (dev-gh) [inline src/components/security_manager/test/security_query_test.cc]: @OHerasym 
I suspect I had fixed that issue already. If you'll have a chance, you can rebase and check.

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 2]
```diff
@@ -36,13 +36,17 @@
 #include "resumption/last_state.h"
 #include "protocol/raw_message.h"
 #include "transport_manager/transport_adapter_listener_mock.h"
-#include "transport_manager/device_mock.h"
+#include "transport_manager/test/include/transport_manager/device_mock.h"
```
Comment (dev-gh) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym 
I would suggest to add appropriate include to make path shorter.

**Comment by dev-gh** [comment]
Reviewed

**Comment by dtrunov** [comment]
Reviewed

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc (hunk 2 above)
Comment (OHerasym) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @dev-gh , b734c8b, Done

File: src/components/security_manager/test/security_query_test.cc (hunk 1 above)
Comment (OHerasym) [inline src/components/security_manager/test/security_query_test.cc]: @dev-gh , I will check it on next pull request to backmerge branch

**Comment by OHerasym** [comment]
@anosach-luxoft , @dev-gh, @VVeremjova , @AByzhynar, @Kozoriz, @LuxoftAKutsan, @dtrunov, Please review changes

**Comment by VVeremjova** [comment]
Reviewed

File: src/components/protocol_handler/test/multiframe_builder_test.cc [hunk 3]
```diff
@@ -42,6 +42,15 @@ namespace test {
 namespace components {
 namespace protocol_handler_test {
 
+namespace {
+const int32_t MICROSECONDS_IN_MILLISECONDS = 1000;
+}
```
Comment (AByzhynar) [inline src/components/protocol_handler/test/multiframe_builder_test.cc]: @OHerasym Please add `// namespace` to the end of this line

File: src/components/protocol_handler/test/protocol_packet_test.cc [hunk 4]
```diff
@@ -205,11 +205,12 @@ TEST_F(ProtocolPacketTest, SetData) {
   EXPECT_EQ(session_id, protocol_packet.data()[3]);
 }
 
-TEST_F(ProtocolPacketTest, DeserializeZeroPacket) {
-  uint8_t message[] = {};
+// TODO {OHerasym} : error C2466: cannot allocate an array of constant size 0
```
Comment (AByzhynar) [inline src/components/protocol_handler/test/protocol_packet_test.cc]: @OHerasym Wrong `TODO` comment style

File: src/components/protocol_handler/test/protocol_payload_test.cc [hunk 5]
```diff
@@ -75,7 +75,7 @@ void prepare_data(uint8_t* data_for_sending, ProtocolPayloadV2& message) {
   if (message.data.size() != 0) {
     uint8_t* current_pointer =
         data_for_sending + offset + message.json.length();
-    u_int32_t binarySize = message.data.size();
+    uint32_t binarySize = message.data.size();
```
Comment (AByzhynar) [inline src/components/protocol_handler/test/protocol_payload_test.cc]: @OHerasym `size_t`?

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 6]
```diff
@@ -113,17 +95,19 @@ TEST_F(TcpAdapterTest, StoreDataWithOneDeviceAndOneApplication) {
...
-  ASSERT_FALSE(tcp_dict["devices"][0]["applications"].isNull());
-  ASSERT_FALSE(tcp_dict["devices"][0]["address"].isNull());
-  EXPECT_EQ(1u, tcp_dict["devices"][0]["applications"].size());
-  EXPECT_EQ(string_port,
-            tcp_dict["devices"][0]["applications"][0]["port"].asString());
-  EXPECT_EQ(uniq_id, tcp_dict["devices"][0]["name"].asString());
+  utils::json::JsonValue dict = last_state_.dictionary();
+  utils::json::JsonValue tcp_dict = dict["TransportManager"]["TcpAdapter"];
+
+  int index = 0;
```
Comment (AByzhynar) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym Why not `const`?

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 7]
```diff
@@ -164,18 +148,19 @@ TEST_F(TcpAdapterTest, StoreDataWithSeveralDevicesAndOneApplication) {
   }
   transport_adapter.CallStore();
 
+  unsigned int index = 0;
```
Comment (AByzhynar) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym `const`?

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 8]
```diff
@@ -243,7 +228,8 @@ TEST_F(TcpAdapterTest, StoreData_ConnectionNotExist_DataNotStored) {
   MockTCPTransportAdapter transport_adapter(
       port, last_state_, transport_manager_settings);
   std::string uniq_id = "unique_device_name";
-  utils::SharedPtr<MockTCPDevice> mockdev = new MockTCPDevice(port, uniq_id);
+  utils::SharedPtr<MockTCPDevice> mockdev =
```
Comment (AByzhynar) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym MakeShared?

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 9]
```diff
@@ -278,7 +266,8 @@ TEST_F(TcpAdapterTest, StoreDataWithOneDevice_RestoreData) {
...
```
Comment (AByzhynar) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym MakeShared?

File: src/components/transport_manager/test/tcp_transport_adapter_test.cc [hunk 10]
```diff
@@ -317,9 +306,9 @@ TEST_F(TcpAdapterTest, StoreDataWithSeveralDevices_RestoreData) {
   std::string uniq_id[count_dev];
   for (uint32_t i = 0; i < count_dev; i++) {
     char numb[12];
-    std::snprintf(numb, 12, "%d", i);
+    snprintf(numb, 12, "%d", i);
     uniq_id[i] = "unique_device_name" + std::string(numb);
-    mockdev[i] = new MockTCPDevice(port, uniq_id[i]);
+    mockdev[i] = new MockTCPDevice(host_address, uniq_id[i]);
```
Comment (AByzhynar) [inline src/components/transport_manager/test/tcp_transport_adapter_test.cc]: @OHerasym MakeShared?

**Comment by AByzhynar** [comment]
Reviewed

**Comment by OHerasym** [comment]
@AByzhynar , All fixed in [this pull request](https://github.com/anosach-luxoft/sdl_core/pull/24) ,  in temp commit2

**Pull request closed without merging**